hack/build/build-go.sh

```diff
@@ -64,6 +64,7 @@ elif [ "${go_opt}" == "build" ]; then
 			ln -sf ${outFile} ${outLink}
 		)
 	done
+    build_func_tests
```

Comment:
I don't think we should unconditionally build tests every time `make build` is run.  I would prefer to see either

 - a new make target to build the func tests
or
- the `make functest` target modified to build and run the tests